flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/nodes/dataset/DataSetAggregate.scala

```diff
@@ -157,4 +161,41 @@ class DataSetAggregate(
       case _ => result
     }
   }
+
+  /**
+    * Dummy [[Row]] into a [[DataSet]] for result after map operations.
+    * @param mapOperator after which insert dummy records
+    * @param tableEnv [[BatchTableEnvironment]] for getting rel builder and type factory
+    * @tparam IN mapOperator input type
+    * @tparam OUT mapOperator output type
+    * @return DataSet of type Row is contains null literals for columns
+    */
+  private def dummyRow[IN,OUT](
+      mapOperator: MapOperator[IN,OUT],
+      tableEnv: BatchTableEnvironment): DataSet[Row] = {
+
+    val builder: RelDataTypeFactory.FieldInfoBuilder = getCluster.getTypeFactory.builder
+    val rowInfo = mapOperator.getResultType.asInstanceOf[RowTypeInfo]
+
+    val nullLiterals :ImmutableList[ImmutableList[RexLiteral]] =
+      ImmutableList.of(ImmutableList.copyOf[RexLiteral](
+        for (fieldName <- rowInfo.getFieldNames)
+          yield {
+            val columnType = tableEnv.getTypeFactory
+              .createTypeFromTypeInfo(rowInfo.getTypeAt(fieldName))
+            builder.add(fieldName, columnType)
+            tableEnv.getRelBuilder.getRexBuilder
+              .makeLiteral(null,columnType,false).asInstanceOf[RexLiteral]
+          }))
+
+    val dataType = builder.build()
+
+    val relNode = RelFactories.DEFAULT_VALUES_FACTORY
+      .createValues(getCluster, dataType, nullLiterals)
+
+    DataSetValuesRule.INSTANCE.asInstanceOf[DataSetValuesRule]
```

Comment:
sorry, I don't understand your point. Could you explain, please?
Do you propose add new `DataSetNullValues extends LogicalAggregate with DataSetRel` for opportunity translate this to `DataSet`? and create rule for initialize new DataSetNullValues. It is correct?

Could I use ` DataSetValues` directly, maybe?  for example
```scala
val relNode = RelFactories.DEFAULT_VALUES_FACTORY
      .createValues(getCluster, dataType, nullLiterals)

    new DataSetValues(
      cluster,
      relNode.getTraitSet.replace(DataSetConvention.INSTANCE),
      dataType,
      nullLiterals,
      "DataSetNullValuesRule"
    ).translateToPlan(tableEnv,Some(TypeConverter.DEFAULT_ROW_TYPE)).asInstanceOf[DataSet[Row]]
```